File: 1403-minimum-subsequence-in-non-increasing-order/1403-minimum-subsequence-in-non-increasing-order.py
class Solution:
    def minSubsequence(self, nums: List[int]) -> List[int]:
        nums, res, cur_sum, sm = list(map(lambda x: -x, nums)), [], 0, sum(nums)
        heapq.heapify(nums)
        while cur_sum <= sm:
            n = heapq.heappop(nums)
            cur_sum += abs(n)
            sm -= abs(n)
            res.append(abs(n))
        return res

File: 1403-minimum-subsequence-in-non-increasing-order/1403-minimum-subsequence-in-non-increasing-order.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var minSubsequence = function(nums) {
    const sum = nums.reduce((acc, val) => acc + val, 0)
    nums.sort((a, b) => b - a)
    const res = []
    let curSum = 0
    for (const n of nums) {
        curSum += n
        res.push(n)
        if (curSum > sum - curSum) break
    }
    return res
};